rust/agama-lib/src/utils/url.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+use url::form_urlencoded::byte_serialize;
+
+pub fn encoded(value: String) -> String {
+    let serialized_value: String = byte_serialize(value.as_bytes()).collect();
+    // Encode space to '%20' as per url standard
+    // Should be fixed by https://github.com/servo/rust-url/pull/1028
+    serialized_value.replace("+", "%20")
+}
```

Comment:
Thanks for this! I think it's more useful to have the encoding logic here rather than buried in a network-related file. While it's true that encoding issues have only bitten us in the network layer so far, it's generally safer to always encode URL query strings to prevent future problems.

That said, I have two minor non-blocking comments:

* I found the alternative solution proposed in [this comment](https://github.com/agama-project/agama/pull/2605#discussion_r2227843048) a bit more straightforward and even sensible, as the code is simpler and not tied to forms. That said, I have no experience with Rust, so I might be missing something.
* I would personally name the function `encode` rather than `encoded`, but again, I’m unfamiliar with Rust naming conventions.

So, overall it LGTM!